s3fs/core.py

```diff
@@ -480,9 +480,26 @@ def _prepare_config_kwargs(self):
 
     async def set_session(self, refresh=False, kwargs={}):
         """Establish S3 connection object.
+
+        ``connect`` is an alias of ``set_session``. This method called with
```

Comment:
This method is called by any operation on an s3fs instance. 

You might call this directly from async code, so that you can assign the client object to a local variable
(^ this is implied in the documentation on async, but should no longer be important for most people).